smac/tae/execute_func.py

```diff
@@ -170,7 +187,17 @@ def run(self, config, instance=None,
                 cost = self.crash_cost
             
             runtime = time.time() - start_time
-            additional_run_info = {}
+
+        # check serializability of results
+        try:
+            json.dumps(cost)
+        except TypeError:
+            raise TypeError("TA returned result of type %s but it should be a serializable type." % type(cost))
+        try:
+            json.dumps(additional_run_info)
+        except TypeError:
+            raise TypeError("TA returned 'additional_run_info' with some non-serializable items. "
+                            "Please ensure all objects returned are serializable.")
```

Comment:
Could you please print the output here, too? Also, could you please somehow add the original error message so that the users sees what kind of object cannot be serialized?

Also, please do not abbreviate the word TA here.

Reply:
Sorry for not being not clear enough here. Issue #453 shows that SMAC complains with the error message `TypeError: Object of type 'int64' is not JSON serializable`. Could you please parse the message out of the exception object and print that, too? Then, the user would know which object cannot be serialized.